### ISSUE TITLE
Generate coloramps.h directly rather than via "sed" on coloramps.inc

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#713]
Generate coloramps.h directly rather than via "sed" on coloramps.inc

This was introduced in MR #573 implementing #402.

As mentioned in https://github.com/madgraph5/madgraph4gpu/pull/573#issuecomment-1356511725
```
By using brute force sed/awk I transformed coloramps.inc into coloramps.h in such a way that they have the same definition of icolamp. Assuming that the one in Fortran is correct (is it?!) then also the one in cudacpp is correct.
```

This is also related to #655: after the latest select_color changes, we no longer get the same LHE color in foratran and cudacpp.

Ideally, eventually we should be able to generate coloramps.h directly rather than via "sed" on coloramps.inc. This is also related to removing patchMad.sh in #656

**Comment by valassi** [comment]
This has been completed in MR #742.

@oliviermattelaer suggested to try and remove my patch that was recreating coloramps.h from coloramps.inc (#713), and use coloramps.h directly, using the results of his previous select_color changes. This turns out to fix the issue in color choices in ggttgg LHE files (fixing #655): a very simple change that fixes two issues at the same time.

**Comment by valassi** [comment]
Note the discussion in #873. I think that this MR #742 fixed the #655 issue for the wrong reasons, i.e. because it introiduced two bugs that cancelled each other ofr iconfig=1:
- there was already a wrong mapping iconfig to channelid
- the MR modified coloramps.h to differe from coloramps.h, because of the underlying (and previously unexposed) bug in icolamp https://github.com/mg5amcnlo/mg5amcnlo/issues/114

Now #873 should normally fix this whole situation...
